File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Reservation.java
package tn.eesprit.gestionevenementback.Entities;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.extern.java.Log;

import java.io.Serializable;
import java.util.Date;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Reservation implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer reservationId;
    Boolean transportIncluded;
    String transportStartingAdress;
    Boolean housingIncluded;
    Boolean seated;
    Float paidPrice;
    Date reservationDate;
    @PrePersist
    private void prePersist() {
        reservationDate = new Date();
    }
    @Enumerated(EnumType.STRING)
    Status status;
    @OneToOne(mappedBy = "reservation",cascade = CascadeType.REMOVE)
    Payement payement;
    @ManyToOne
    User user;
    @JsonIgnore
    @ManyToOne
    Event event;
    @ManyToOne(cascade = CascadeType.PERSIST)
    Transport transport;
}

File: GestionEvenementFront/src/app/_Models/Reservation.ts
import {Payement} from "./Payement";
import {User} from "./User";
import {Transport} from "./Transport";
import {Status} from "./Status";
import {Event} from "./Event";

export class Reservation {
  reservationId:number;
  transportIncluded:boolean;
  transportStartingAdress:string;
  housingIncluded:boolean;
  paidPrice:number;
  payement:Payement;
  event:Event ;
  user:User;
  seated:boolean;
  transport:Transport;
  status:Status;
  edit:boolean
  reservationDate:Date;
}
